### ISSUE TITLE
Does vcpkg store its installation path? 

### ISSUE THREAD

**Issue by Krzmbrzl** [microsoft/vcpkg#13269]
I installed vcpkg in a directory on my C drive (I'm on Windows) but it turned out that I hadn't enough space left so after vcpkg had crashed on me, I moved the directory to my D drive and resumed the installation.

In my project that points to the vcpkg toolchains cmake file on the D drive, cmake now always references the directory on the C drive for the brotli library and as that directory no longer exists,the build fails. 

In the meantime I have completely deleted the vcpkg directory and the project's directory and cloned them new (and also installed the build env via vcpkg anew) but the error remains the same. 

Thus I am wondering whether vcpkg (or brotli is particular) stores its location somewhere external? There's no other way I can think of that cmake would reference the wrong vcpkg directory 🤔

I tried searching the registry and I also deleted all vcpkg related files from Appdata but still no success. 

Any ideas?

**Comment by Krzmbrzl** [comment]
Okay I searched again and this time I did find references to the old directory in the package-config (`*.pc`) files. Any idea how they might end up using the wrong path?

**Comment by Neumann-A** [comment]
> Okay I searched again and this time I did find references to the old directory in the package-config (*.pc) files. Any idea how they might end up using the wrong path?

somebody missed the call to `vcpkg_fixup_pkgconfig()`. Not all ports with install pc files have yet been updated to call that function. 
The same happens if a `vcpkg_fixup_cmake_targets()` call is missing. 

Did you delete your cmake cache after moving vcpkg? Some things can only be set on the first configure call.

**Comment by Krzmbrzl** [comment]
> Did you delete your cmake cache after moving vcpkg? Some things can only be set on the first configure call.

I removed the entire vcpkg directory and cloned vcpkg from scratch but the problem persisted.


Explicitly setting `-DCMAKE_BUILD_TYPE=Release` solved the issue somehow. Using `Debug` results in the mentioned error (as does not explicitly setting the build type even though theoretically our project should use Release as a default)

**Comment by Neumann-A** [comment]
> Using Debug results in the mentioned error

than you have to check the `*.pc` file in `debug/lib/pkgconfig`

**Comment by Krzmbrzl** [comment]
> than you have to check the *.pc file in debug/lib/pkgconfig

It has almost the exact same content:
```
prefix=${pcfiledir}/../..
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/../include

Name: libbrotlicommon
URL: https://github.com/google/brotli
Description: Brotli common dictionary library
Version: 1.0.7
Libs: -L${libdir} -lbrotlicommon-static
Cflags: -I${includedir}
```

**Comment by Krzmbrzl** [comment]
> somebody missed the call to vcpkg_fixup_pkgconfig(). Not all ports with install pc files have yet been updated to call that function.
The same happens if a vcpkg_fixup_cmake_targets() call is missing.

But how can this still reference the old vcpkg directory (that no longer exists) when vcpkg has been completely deleted and cloned again? There were no files left over from the build on the C: drive in that case :eyes:

**Comment by Neumann-A** [comment]
ok I repeat myself:
> Did you delete your cmake cache after moving vcpkg? Some things can only be set on the first configure call.

this has nothing to do with vcpkg. The question here was: did you make a clean configure call of your project after moving it. (Meaning deleting the CMake cache. )

**Comment by Krzmbrzl** [comment]
> this has nothing to do with vcpkg. The question here was: did you make a clean configure call of your project after moving it. (Meaning deleting the CMake cache. )

Yeah that as well. I tried removing all files in the build directory, removing the entire build directory and eventually removing the entire project directory (my project in this case - not vcpkg) and clone it again. The result stays the same.

**Comment by Krzmbrzl** [comment]
I just noticed though that it seems that `vcpkg` has been installed using
```
vcpkg integrate install
```
What exactly does this do and could this be responsible for old paths being used?

**Comment by Krzmbrzl** [comment]
And just to give all information - I used [this script](https://github.com/mumble-voip/mumble-releng-vcpkg/blob/master/Get-MumbleDeps.ps1) with a custom `$vcpkgdir` value to install the dependencies.

And this is my cmake call:
```
cmake -G "Ninja" "-DVCPKG_TARGET_TRIPLET=x64-windows-static-md" "-Dstatic=ON" "-DCMAKE_TOOLCHAIN_FILE=D:/mumble-vcpkg/scripts/buildsystems/vcpkg.cmake" "-DIce_HOME=D:/mumble-vcpkg/installed/x64-windows-static-md" "-DCMAKE_BUILD_TYPE=Release" ..
```

I also tried using  `-G "NMake Makefiles` but that didn't make a difference

**Comment by PhoebeHui** [comment]
@Krzmbrzl, do you create symlik? what output does 'dir ./vcpkg/ports/ brotli' show?

**Comment by PhoebeHui** [comment]
BTW, VCPKG doesn't store the path, and I can't repro this issue locally.

**Comment by Krzmbrzl** [comment]
> do you create symlik?

No I haven't used any symlinks (nor other links)

>  what output does 'dir ./vcpkg/ports/ brotli' show?
```
    Directory: D:\mumble-vcpkg\ports


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----       31.08.2020     19:16                brotli
```

**Comment by Krzmbrzl** [comment]
Okay I reinstalled vcpkg and all our packages again (manually and one package at a time in order to track down which package causes the wrong paths to be used) and for some reason it seems to be working just fine now.

My theory is that either a restart between the vcpkg installations was required or that a custom port file that we're using somehow messes things up (I have excluded it from my current test until now).

Whatever the cause was, it seems to be gone now. Thanks for having looked into tshi with me :)
